src/nvim/tui/tui.c

```diff
@@ -910,12 +910,23 @@ static void fix_terminfo(TUIData *data)
   unibi_set_if_empty(ut, unibi_clr_eos, "\x1b[J");
 }
 
-static void flush_buf(UI *ui)
+static void flush_buf_raw(UI *ui)
 {
   uv_write_t req;
   uv_buf_t buf;
   TUIData *data = ui->data;
 
+  buf.base = data->buf;
+  buf.len = data->bufpos;
+  uv_write(&req, (uv_stream_t *)&data->output_handle, &buf, 1, NULL);
+  uv_run(&data->write_loop, UV_RUN_DEFAULT);
+  data->bufpos = 0;
+}
+
+static void flush_buf(UI *ui)
+{
+  TUIData *data = ui->data;
+
   if (!data->busy) {
```

Comment:
Since the cursor fiddling is already guarded by a boolean check, we could add a `bool toggle_cursor` parameter to `flush_buf`. IMO that's less awkward than having a "raw" variant of the function.